internal/profile/install_profile.go

```diff
@@ -0,0 +1,212 @@
+func CreateProfileFrom(elasticPackagePath string, newProfileName string, fromProfileName string) error {
+
+	fromProfile, err := LoadProfile(elasticPackagePath, fromProfileName)
+	if err != nil {
+		return errors.Wrapf(err, "error loading %s profile", fromProfileName)
+	}
+
+	newProfile, err := NewConfigProfile(elasticPackagePath, newProfileName)
+	if err != nil {
+		return errors.Wrapf(err, "error creating %s profile", newProfileName)
+	}
+
+	newExists, err := newProfile.profileAlreadyExists()
+	if err != nil {
+		return errors.Wrapf(err, "error checking profile %s", newProfile.ProfilePath)
+	}
+	if newExists {
+		return errors.Errorf("profile %s already exists", newProfile.profileName)
+	}
+	os.Mkdir(newProfile.ProfilePath, 0755)
+
+	newProfile.updateFileBodies(fromProfile.configFiles)
+	return newProfile.writeProfileResources()
+
+}
+
+// LoadProfileFromDefaultLocation loads an existing profile from the default elastic-package config dir
+func LoadProfileFromDefaultLocation(profileName string) (*ConfigProfile, error) {
+
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding stack dir location")
+	}
+
+	return LoadProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfileFromDefaultLocation deletes a profile from the default elastic-package config dir
+func DeleteProfileFromDefaultLocation(profileName string) error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return DeleteProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfile deletes a given config profile.
+func DeleteProfile(elasticPackagePath string, profileName string) error {
+
+	if profileName == DefaultProfile {
+		return errors.New("cannot remove default profile")
+	}
+
+	pathToDelete := filepath.Join(elasticPackagePath, profileName)
+
+	return os.RemoveAll(pathToDelete)
+
+}
+
+// PrintProfilesFromDefaultLocation lists known packages in the default elastic-package install dir
+func PrintProfilesFromDefaultLocation() error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return PrintProfiles(loc.StackDir())
+}
+
+// PrintProfiles lists known profiles
+func PrintProfiles(elasticPackagePath string) error {
+
+	profiles, err := FetchAllProfiles(elasticPackagePath)
+	if err != nil {
+		return errors.Wrap(err, "error fetching profiles")
+	}
+
+	//print later, after we've run into any errors, to avoid some ugly-half printing if something fails halfway through
+	header := []string{"Name", "Date Created", "User", "elastic-package version", "Path"}
+	for _, headerVal := range header {
+		fmt.Printf("%-30s ", headerVal)
+	}
+	fmt.Printf("\n")
+	for _, iter := range profiles {
+		fmt.Printf("%-30s %-30s %-30s %-30s %-30s\n", iter.Name, iter.DateCreated.Format(time.RFC3339), iter.User, iter.Version, iter.From)
```

Comment:
Could you please move all the printing to the `cmd` level and use internal just to read/return available profiles?